Marlin/Marlin_main.cpp

```diff
@@ -6840,9 +6837,7 @@ inline void gcode_M503() {
 
     #if IS_KINEMATIC
       // Move XYZ to starting position, then E
-      inverse_kinematics(lastpos);
-      planner.buffer_line(delta[A_AXIS], delta[B_AXIS], delta[C_AXIS], destination[E_AXIS], FILAMENT_CHANGE_XY_FEEDRATE, active_extruder);
-      planner.buffer_line(delta[A_AXIS], delta[B_AXIS], delta[C_AXIS], lastpos[E_AXIS], FILAMENT_CHANGE_XY_FEEDRATE, active_extruder);
+      planner.buffer_line_kinematic(lastpos, FILAMENT_CHANGE_XY_FEEDRATE, active_extruder);
```

Comment:
the original code is following:

    // Set extruder to saved position
    current_position[E_AXIS] = lastpos[E_AXIS];
    destination[E_AXIS] = lastpos[E_AXIS];
    planner.set_e_position_mm(current_position[E_AXIS]);

    #if IS_KINEMATIC
      // Move XYZ to starting position, then E
      inverse_kinematics(lastpos);
      planner.buffer_line(delta[A_AXIS], delta[B_AXIS], delta[C_AXIS], destination[E_AXIS], FILAMENT_CHANGE_XY_FEEDRATE, active_extruder);
      planner.buffer_line(delta[A_AXIS], delta[B_AXIS], delta[C_AXIS], lastpos[E_AXIS], FILAMENT_CHANGE_XY_FEEDRATE, active_extruder);
    #else
      // Move XY to starting position, then Z, then E
      destination[X_AXIS] = lastpos[X_AXIS];
      destination[Y_AXIS] = lastpos[Y_AXIS];
      RUNPLAN(FILAMENT_CHANGE_XY_FEEDRATE);
      destination[Z_AXIS] = lastpos[Z_AXIS];
      RUNPLAN(FILAMENT_CHANGE_Z_FEEDRATE);
    #endif

the only difference between the two buffer_line calls are destination[E_AXIS} and lastpos[E_AXIS], but few lines before is destination[E_AXIS] = lastpos[E_AXIS];
So the last buffer_line does not move the extruder.
In the other branch there are 2 moves - first to x/y, second to z. No change for extruder.
So I think, that the comment is wrong. The words "then E" should be removed.

I've modified the comments, I'll push the patch